agent/http_test.go

```diff
@@ -786,6 +787,97 @@ func TestEnableWebUI(t *testing.T) {
 	}
 }
 
+func TestParseToken_ProxyTokenResolve(t *testing.T) {
+	t.Parallel()
+
+	type endpointCheck struct {
+		endpoint string
+		handler  func(s *HTTPServer, resp http.ResponseWriter, req *http.Request) (interface{}, error)
```

Comment:
Huh TIL type of a method can be expressed this way with receiver as first arg. I knew that receivers are just syntax sugar on that generally but not that this was possible...

Reply:
Yeah, I learned that by reading how we register HTTP endpoints because we do the same thing there.